Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -226,7 +231,39 @@ public final class CRTAWSCredentialsProvider {
         return future
     }
 
+    static func setUpShutDownOptions(shutDownOptions: CRTCredentialsProviderShutdownOptions?)
+    -> aws_credentials_provider_shutdown_options {
+
+        let pointer: UnsafeMutablePointer<CRTCredentialsProviderShutdownOptions>? = fromOptionalPointer(ptr: shutDownOptions)
+        let shutDownOptionsC = aws_credentials_provider_shutdown_options(shutdown_callback: { userData in
+            guard let userData = userData else {
+                return
+            }
+            let pointer = userData.assumingMemoryBound(to: CRTCredentialsProviderShutdownOptions.self)
+            defer {pointer.deinitializeAndDeallocate()}
```

Comment:
After the `shutDownCallback` call, you can just do this without a defer, right?

Reply:
correct, updated code